### PR TITLE
feat: added check_factorial

### PR DESCRIPTION
**Pull request by divyajyotiuk**
#### Description of Change
Added a check_factorial algorithm under math.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1155"></a>

### CODE REVIEW

**Comment by mishraabhinn** [comment]
[C-Plus-Plus/math/fibonacci.cpp
](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/math/fibonacci.cpp)
I think this repo already has factorial program. 
Is you program something different @divyajyotiuk

**Comment by divyajyotiuk** [comment]
I have added a program that checks whether a number is a factorial of some
number or not. Returns true if it is a factorial of some number, else false.

**Comment by mishraabhinn** [comment]
Actually I just checked the name of the program and thought it was about the find factorial. That why commented. Little misunderstanding.

**Review by mishraabhinn — APPROVED** [review]

**Review by mishraabhinn — COMMENTED** [review]
@kvedala Have a look. Its seems fine to me.

File: math/check_factorial.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        n = n / i;
+    }
+    if (n == 1) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/** Test function */
```
Comment (kvedala) [inline math/check_factorial.cpp:33]:
```suggestion
/** Test function 
 * @returns void
 */
```

File: math/check_factorial.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    std::cout << "Test 4:\t n=479001600\n";
+    assert(is_factorial(479001600) == true);
+    std::cout << "passed\n";
+
+    std::cout << "Test 5:\t n=-24\n";
+    assert(is_factorial(-24) == false);
+    std::cout << "passed\n";
+}
+
+/** Main function */
```
Comment (kvedala) [inline math/check_factorial.cpp:56]:
```suggestion
/** Main function 
 * @returns 0 on exit
 */
```

File: math/check_factorial.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/**
+ * @file
+ * A simple program to check if the given number is a factorial of some number
```
Comment (kvedala) [inline math/check_factorial.cpp:3]:
```suggestion
 * @brief A simple program to check if the given number is a factorial of some number
```

**Review by kvedala — CHANGES_REQUESTED** [review]
please undo the changes to `.vscode/settings.json` and make the requested changes

File: math/check_factorial.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+#include <cassert>
+#include <iostream>
+
+/**
+ * Function to check if the given number is factorial of some number or not.
+ * @param n number to be checked.
+ * @return if number is a factorial, returns true, else false.
+ */
+
+bool is_factorial(int n) {
```
Comment (kvedala) [inline math/check_factorial.cpp:16]: wouldn't it be better to use the fixed number type as below?
```suggestion
bool is_factorial(uint64_t n) {
```
similarly, the variable `i` on line 20 could probably be `uint32_t`?

Comment (divyajyotiuk) [reply math/check_factorial.cpp]: Yes, will make the changes. Will help dealing with large numbers.

File: math/check_factorial.cpp [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * Function to check if the given number is factorial of some number or not.
+ * @param n number to be checked.
+ * @return if number is a factorial, returns true, else false.
+ */
+
+bool is_factorial(uint64_t n) {
+    if (n <= 0) {
+        return false;
+    }
+    for (uint32_t i = 1;; i++) {
```
Comment (mishraabhinn) [inline math/check_factorial.cpp:20]: Why uint32_t here and uint64_t in parameter of function. Can't both be uint64_t @kvedala ?

Comment (kvedala) [reply math/check_factorial.cpp:20]: sure, they can. the logic is that the factorial result will overflow the `uint64_t` much early on. the variable `i` would not even get to limit of uint8_t.

Comment (mishraabhinn) [reply math/check_factorial.cpp:20]: I got your point.
but then variable  `i` should also be uint64_t

Comment (kvedala) [reply math/check_factorial.cpp:20]: it can be but doesn't have to be. :)

**Review by kvedala — APPROVED** [review]

**Pull request merged**